packages/server-wallet/src/message-service/test-message-service.ts

```diff
@@ -44,20 +47,80 @@ export class TestMessageService implements MessageServiceInterface {
    * @returns
    */
   protected constructor(handleMessage: MessageHandler, protected _logger?: Logger) {
+    super();
     this._options = {dropRate: 0, meanDelay: undefined};
     this._abortController = new AbortController();
     this._handleMessages = async messages => {
       for (const message of messages) {
-        // This prevents triggering messages after the service is destroyed
-        if (!this._destroyed) return handleMessage(message, this);
+        if (!this._destroyed) {
+          await handleMessage(message);
+        }
       }
     };
   }
 
-  static async create(
-    incomingMessageHandler: MessageHandler,
+  static setLatencyOptions(peerWallets: TestPeerWallets, options: Partial<LatencyOptions>): void {
+    const messageServices = [peerWallets.a.messageService, peerWallets.b.messageService];
+
+    for (const messageService of messageServices) {
+      if (!isTestMessageService(messageService)) {
+        throw new Error('Can only set latency options on a TestMessageService');
+      } else {
+        messageService.setLatencyOptions(options);
+      }
+    }
+  }
+
+  static unfreeze(peerWallets: TestPeerWallets): void {
+    const messageServices = [peerWallets.a.messageService, peerWallets.b.messageService];
+
+    for (const messageService of messageServices) {
+      if (!isTestMessageService(messageService)) {
+        throw new Error('Can only set latency options on a TestMessageService');
+      } else {
+        messageService.unfreeze();
+      }
+    }
+  }
+
+  static freeze(peerWallets: TestPeerWallets): void {
+    const messageServices = [peerWallets.a.messageService, peerWallets.b.messageService];
+
+    for (const messageService of messageServices) {
+      if (!isTestMessageService(messageService)) {
+        throw new Error('Can only set latency options on a TestMessageService');
+      } else {
+        messageService.freeze();
+      }
+    }
+  }
+  static linkMessageServices(
+    messageService1: MessageServiceInterface,
+    messageService2: MessageServiceInterface,
     logger?: Logger
-  ): Promise<TestMessageService> {
+  ): void {
+    if (!isTestMessageService(messageService1) || !isTestMessageService(messageService2)) {
+      throw new Error('Cannot link message services besides the TestMessageService');
+    }
+
+    messageService1.on('deliveryRequested', async (messages: Message[]) => {
+      logger?.trace(
+        {messages: messages.map(m => formatMessageForLogger(m))},
```

Comment:
```suggestion
        {messages: messages.map(formatMessageForLogger)},
```

Reply:
It'd be nice if there was a lint rule for this.